web/packages/teleterm/src/ui/ModalsHost/ModalsHost.tsx

```diff
@@ -132,6 +133,24 @@ function renderDialog(dialog: Dialog, handleClose: () => void) {
       );
     }
 
+    case 'headless-authn': {
+      return (
+        <HeadlessAuthentication
+          rootClusterUri={dialog.rootClusterUri}
+          headlessAuthenticationId={dialog.headlessAuthenticationId}
+          clientIp={dialog.headlessAuthenticationClientIp}
+          onCancel={() => {
+            handleClose();
+            dialog.onCancel?.();
```

Comment:
`onCancel` wasn't marked as optional in the `DialogHeadlessAuthentication` interface, so we don't need an optional chaining here.
```suggestion
            dialog.onCancel();
```